src/ccmain/thresholder.cpp

```diff
@@ -205,22 +204,19 @@ std::tuple<bool, Image, Image, Image> ImageThresholder::Threshold(
     return std::make_tuple(false, nullptr, pix_binary, nullptr);
   }
 
-  pix_grey = GetPixRectGrey();
-
-  if (method == ThresholdMethod::Otsu || method >= ThresholdMethod::Max) {
-    method = ThresholdMethod::AdaptiveOtsu;
-  }
+  auto pix_grey = GetPixRectGrey();
 
   int r;
-  if (method == ThresholdMethod::AdaptiveOtsu) {
-    r = pixOtsuAdaptiveThreshold(pix_grey, 300, 300, 0, 0, 0.1,
-                                 pix_thresholds, pix_binary);
-  } else if (method == ThresholdMethod::TiledSauvola) {
+  if (method == ThresholdMethod::TiledSauvola) {
     r = pixSauvolaBinarizeTiled(pix_grey, 25, 0.40, 300, 300, pix_thresholds,
                                 pix_binary);
+  } else {
+    // AdaptiveOtsu.
+    r = pixOtsuAdaptiveThreshold(pix_grey, 300, 300, 0, 0, 0.1,
+                                 pix_thresholds, pix_binary);
   }
 
-  bool ok = r == 0 ? true : false;
```

Comment:
I intend to add more options from Leptonica, but I guess I can keep this the last option.

Reply:
That should not be a problem. You can also choose any order of the different methods. Just make sure that the last block is executed unconditionally. Then the compilers are happy.